src/main/scala/temple/generate/server/go/auth/GoAuthServiceDAOGenerator.scala

```diff
@@ -1,9 +1,189 @@
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.generate.utils.CodeUtils
+import temple.utils.StringUtils.{doubleQuote, tabIndent}
+
+import scala.collection.immutable.ListMap
 
 object GoAuthServiceDAOGenerator {
 
-  private[auth] def generateErrors(): String =
-    FileUtils.readResources("go/genFiles/auth/dao/errors.go.snippet").stripLineEnd
+  private[auth] def generateImports(root: AuthServiceRoot): String =
+    mkCode(
+      "import",
+      CodeWrap.parens.tabbed(
+        doubleQuote("database/sql"),
+        doubleQuote("fmt"),
+        "",
+        doubleQuote(s"${root.module}/util"),
+        doubleQuote("github.com/google/uuid"),
+        "",
+        "// pq acts as the driver for SQL requests",
+        doubleQuote("github.com/lib/pq"),
+      ),
+    )
+
+  private[auth] def generateGlobals(): String =
+    mkCode.lines(
+      "// https://www.postgresql.org/docs/9.3/errcodes-appendix.html",
+      genConst("psqlUniqueViolation", doubleQuote("unique_violation")),
+    )
+
+  private[auth] def generateInterface(): String =
+    mkCode.lines(
+      "// Datastore provides the interface adopted by the DAO, allowing for mocking",
+      mkCode(
+        "type Datastore interface",
+        CodeWrap.curly.tabbed(
+          "CreateAuth(input CreateAuthInput) (*Auth, error)",
+          "ReadAuth(input ReadAuthInput) (*Auth, error)",
+        ),
+      ),
+    )
+
+  private[auth] def generateStructs(root: AuthServiceRoot): String = {
+    val id       = ListMap(root.idAttribute.name.toUpperCase           -> generateGoType(root.idAttribute.attributeType))
+    val auth     = ListMap(root.authAttribute.authType.name.capitalize -> generateGoType(root.authAttribute.attributeType))
+    val password = ListMap("Password"                                  -> "string")
+    mkCode.lines(
+      "// Auth encapsulates the object stored in the datastore",
+      genStruct("Auth", id ++ auth ++ password),
+      "",
+      "// CreateAuthInput encapsulates the information required to create a single auth in the datastore",
+      genStruct("CreateAuthInput", id ++ auth ++ password),
+      "",
+      "// ReadAuthInput encapsulates the information required to read a single auth in the datastore",
+      genStruct("ReadAuthInput", auth),
+    )
+  }
+
+  private[auth] def generateQueryFunctions(): String =
+    mkCode.doubleLines(
+      GoCommonDAOGenerator.generateExecuteQueryWithRowResponse(),
+      GoCommonDAOGenerator.generateExecuteQuery(),
+    )
+
+  private def generateCreateFunction(root: AuthServiceRoot, scanFunctionCall: String): String = {
+    val queryStatement = genDeclareAndAssign(
+      genFunctionCall(
+        "executeQueryWithRowResponse",
+        "dao.DB",
+        doubleQuote(
+          s"INSERT INTO auth (${root.idAttribute.name}, ${root.authAttribute.authType.name}, password) VALUES ($$1, $$2, $$3) RETURNING id, name, password",
+        ),
+        s"input.${root.idAttribute.name.toUpperCase()}",
```

Comment:
```suggestion
        s"input.${root.idAttribute.name.toUpperCase}",
```
feels weirdly inconsistent with the line below